syz-verifier/main.go

```diff
@@ -222,13 +221,33 @@ func main() {
 	select {}
 }
 
+func getEnabledCalls(target *prog.Target, enabled []int) map[*prog.Syscall]bool {
+	calls := make(map[*prog.Syscall]bool)
+	for _, id := range enabled {
+		calls[target.Syscalls[id]] = true
+	}
+
+	// Find and report to the user all the system calls that need to be
+	// disabled due to missing dependencies.
+	_, disabled := target.TransitivelyEnabledCalls(calls)
```

Comment:
This needs to be done after we check both kernels for supported syscalls. Otherwise we will end up with an inconsistent set of syscalls. If a kernel says it does not support a particular openat variation, we will end up with orphaned read/write syscalls for that fd type.